### PR TITLE
Check for invalid rollback

### PR DESCRIPTION
**Pull request by aboba**
Fix for Issue https://github.com/w3c/webrtc-pc/issues/1470

Related JSEP PR:  https://rawgit.com/w3c/webrtc-pc/master/webrtc.html

***
[Preview](https://s3.amazonaws.com/pr-preview/w3c/webrtc-pc/issue-1470-patch.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/webrtc-pc/a12bf47...e70dc31.html)

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -1301,6 +1301,14 @@
                     <code>InvalidModificationError</code> and abort these steps.
                   </li>
                   <li>
+                    <p>If <var>description</var> is set as a local description,
+                    if <code><var>description</var>.type</code> is <code>"rollback"</code>
+                    and <a>signaling state</a> is not <code>"have-local-offer"</code>
```
Comment (jan-ivar) [inline webrtc.html]: Maybe *"is neither "have-local-offer" nor "have-local-pranswer""* ?

I'm not a JSEP expert, but [section 4.1.8.2.](https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-20#section-4.1.8.2) seems to imply rollback is allowed from any state but stable.

File: webrtc.html [hunk 2]
```diff
@@ -1392,6 +1400,14 @@
                     <p>Otherwise, if <var>description</var> is set as a remote
                     description, then run one of the following steps:</p>
                     <ul>
+                      <li>
+                        <p>If <var>description</var> is set as a remote description,
+                        if <code><var>description</var>.type</code> is <code>"rollback"</code>
+                        and <a>signaling state</a> is not <code>"have-remote-offer"</code>
```
Comment (jan-ivar) [inline webrtc.html]: *"is neither "have-remote-offer" nor "have-remote-pranswer""* ?

File: webrtc.html [hunk 3]
```diff
@@ -1392,6 +1400,14 @@
                     <p>Otherwise, if <var>description</var> is set as a remote
                     description, then run one of the following steps:</p>
                     <ul>
+                      <li>
+                        <p>If <var>description</var> is set as a remote description,
```
Comment (soareschen) [inline webrtc.html]: This should also be in step 2.2 where it covers _"If the process to apply description fails for any reason..."_.

**Comment by taylor-b** [comment]
I agree with Jan-Ivar. This should be covered by "If the description's type is invalid for the current signaling state of connection, then reject p". If the issue is that JSEP isn't clear enough about what's invalid, then we should fix JSEP. Here's an issue (I made a PR too): https://github.com/rtcweb-wg/jsep/issues/760

**Comment by fluffy** [comment]
See my comments in bug ... I'm not sure this is the behavior we want - I think we need to resolve that in JSEP - but if this is the behavior we decide on there, then of source we need something like this.

**Comment by taylor-b** [comment]
The JSEP decision was "you can roll back PR-answers, and it makes no difference whether you call SLD or SRD". So this PR should be updated. The only invalid rollback is a rollback from a "stable" state.

**Comment by stefhak** [comment]
LGTM.

**Pull request merged**